framework/src/actions/AutoCheckpointAction.C

```diff
@@ -0,0 +1,31 @@
+#include "AutoCheckpointAction.h"
+#include "FEProblem.h"
+#include "Checkpoint.h"
+registerMooseAction("MooseApp", AutoCheckpointAction, "auto_checkpoint_action");
+
+InputParameters
+AutoCheckpointAction::validParams()
+{
+  InputParameters params = Action::validParams();
+
+  return params;
+}
+
+AutoCheckpointAction::AutoCheckpointAction(const InputParameters & params) : Action(params) {}
+
+void
+AutoCheckpointAction::act()
+{
+  if (_app.isUltimateMaster())
+  {
+    if (!_app.getOutputWarehouse().getOutputs<Checkpoint>().empty())
+      // if there's already a checkpoint object, we don't need to worry about creating a new
```

Comment:
Not quite: This will result in some silent odd behavior. If somebody turns on checkpoints and dials back the interval to only output every 10 or 100 steps, then they run on the cluster, they might find that signal handler just doesn't work with no indication why (because we won't have a Checkpoint object with the right attributes set).

I'm thinking that a better design would be to go ahead and turn on auto save if Checkpoint is already enabled.

Reply:
I am a bit confused, I believe this is doing what you are describing. 

The check `!_app.getOutputWarehouse().getOutputs<Checkpoint>().empty()` doesn't check if an existing Checkpoint object has been _written out to yet,_ just that the Checkpoint object has been created internally. The behavior you describe shouldn't be possible.

If you're concerned I could add a test that verifies this behavior. Since the autosave checkpoints write to directories with `autosave_cp` in the name (see my reply a few comments down for more info on this), then all we would have to do is run a test with a regular Checkpoint with a long interval in the test spec, then verify that it doesn't create an autosave checkpoint.

Reply:
What I was trying to say is that you don’t want to short circuit here. If a user adds a CheckPoint object manually through the input file. You aren’t adding the auto save one. 

I think the right solution would be to modify the instance already present to make it auto save in addition to the interval based method.